msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/AccountState.kt

```diff
@@ -227,42 +230,79 @@ class AccountState private constructor(
      * @return [com.microsoft.identity.nativeauth.statemachine.results.GetAccessTokenResult] The result of the getAccessToken action
      */
     suspend fun getAccessToken(forceRefresh: Boolean = false): GetAccessTokenResult {
+        return getAccessTokenInternal(forceRefresh, null);
+    }
+
+    /**
+     * Retrieves the access token for the currently signed in account from the cache.
+     * If the access token is expired, it will be attempted to be refreshed using the refresh token that's stored in the cache;
+     * Kotlin coroutines variant.
+     *
+     * @return [com.microsoft.identity.nativeauth.statemachine.results.GetAccessTokenResult] The result of the getAccessToken action
+     */
+    suspend fun getAccessToken(forceRefresh: Boolean = false, scopes: List<String>): GetAccessTokenResult {
+        var scopeSet = HashSet<String>()
+        if (!scopes.isNullOrEmpty()) {
+            scopeSet.addAll(scopes)
+        }
+        val scopeSet2 = BaseController.addDefaultScopes(scopeSet)
+        return getAccessTokenInternal(forceRefresh, ArrayList(scopeSet2))
+    }
+
+    suspend fun getAccessTokenInternal(forceRefresh: Boolean, scopes: List<String>?): GetAccessTokenResult {
         LogSession.logMethodCall(
             tag = TAG,
             correlationId = null,
             methodName = "$TAG.getAccessToken(forceRefresh: Boolean)"
         )
         return withContext(Dispatchers.IO) {
-            val account =
-                NativeAuthPublicClientApplication.getCurrentAccountInternal(config) as? Account
+
+            val accountResult = NativeAuthPublicClientApplication.getCurrentAccountInternal(config) as? Account
+            val account = accountResult
                     ?: return@withContext GetAccessTokenError(
                         errorType = GetAccessTokenErrorTypes.NO_ACCOUNT_FOUND,
                         error = MsalClientException.NO_CURRENT_ACCOUNT,
                         errorMessage = MsalClientException.NO_CURRENT_ACCOUNT_ERROR_MESSAGE,
                         correlationId = "UNSET"
                     )
 
-            val acquireTokenSilentParameters = AcquireTokenSilentParameters.Builder()
+            val acquireTokenSilentParametersBuilder = AcquireTokenSilentParameters.Builder()
                 .forAccount(account)
                 .fromAuthority(account.authority)
-                .build()
+                .forceRefresh(forceRefresh)
+
+            if (scopes != null ) {
+                acquireTokenSilentParametersBuilder.withScopes(scopes)
+            }
+
+            val acquireTokenSilentParameters = acquireTokenSilentParametersBuilder.build()
 
             val accountToBeUsed = PublicClientApplication.selectAccountRecordForTokenRequest(
```

Comment:
maybe I'm overlooking it, but where is this `accountToBeUsed` param used further in the flow?

Reply:
updated